mobile/android/android-device-file-system.ts

```diff
@@ -53,26 +54,26 @@ export class AndroidDeviceFileSystem implements Mobile.IDeviceFileSystem {
 	public async transferFiles(deviceAppData: Mobile.IDeviceAppData, localToDevicePaths: Mobile.ILocalToDevicePathData[]): Promise<void> {
 		// TODO: Do not start all promises simultaneously as this leads to error EMFILE on Windows for too many opened files.
 		// Use chunks (for example on 100).
-		await Promise.all(
-			_(localToDevicePaths)
-				.filter(localToDevicePathData => this.$fs.getFsStats(localToDevicePathData.getLocalPath()).isFile())
-				.map(async localToDevicePathData => {
-					const devicePath = localToDevicePathData.getDevicePath();
-					await this.adb.executeCommand(["push", localToDevicePathData.getLocalPath(), devicePath]);
-					await this.adb.executeShellCommand(["chmod", "0777", path.dirname(devicePath)]);
-				}
-				)
-				.value()
-		);
-
-		await Promise.all(
-			_(localToDevicePaths)
-				.filter(localToDevicePathData => this.$fs.getFsStats(localToDevicePathData.getLocalPath()).isDirectory())
-				.map(async localToDevicePathData =>
-					await this.adb.executeShellCommand(["chmod", "0777", localToDevicePathData.getDevicePath()])
-				)
-				.value()
-		);
+		const directoriesToChmod: string[] = [];
+		const action = async (localToDevicePathData: Mobile.ILocalToDevicePathData) => {
+			const fstat = this.$fs.getFsStats(localToDevicePathData.getLocalPath());
+			if (fstat.isFile()) {
+				const devicePath = localToDevicePathData.getDevicePath();
+				await this.adb.executeCommand(["push", localToDevicePathData.getLocalPath(), devicePath]);
+				await this.adb.executeShellCommand(["chmod", "0777", path.dirname(devicePath)]);
+			} else if (fstat.isDirectory()) {
+				const dirToChmod = localToDevicePathData.getDevicePath();
+				directoriesToChmod.push(dirToChmod);
+			}
+		};
+
+		await executeActionByChunks<Mobile.ILocalToDevicePathData>(localToDevicePaths, 100, action);
+
+		const dirsChmodAction = async (directoryToChmod: string) => {
+			await this.adb.executeShellCommand(["chmod", "0777", directoryToChmod]);
```

Comment:
this can be safely returned instead of awaited:
```TypeScript
const dirsChmodAction = (directoryToChmod: string) => this.adb.executeShellCommand(["chmod", "0777", directoryToChmod]);
```